components/org.wso2.carbon.identity.local.auth.emailotp/src/main/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticator.java

```diff
@@ -952,6 +954,14 @@ private void triggerEvent(String eventName, AuthenticatedUser user,
                 }
             }
         }
+        // For JIT provisioned federated users add the locale from the mapped user to event properties.
+        if (context != null && user.isFederatedUser()) {
+            String localUserName = getMappedLocalUsername(user, context);
+            if (StringUtils.isNotEmpty(localUserName)) {
+                properties.put(AuthenticatorConstants.LOCAL_CLAIM_VALUE,
```

Comment:
Lets check resolved local value is not empty and put to the list. Otherwise NPE can be occur when utilizing this value.